### PR TITLE
PORTALS-2610: pass through generic card configuration into custom Dat…

### PR DESCRIPTION
**Pull request by jay-hodgson**
…aset row renderer, and respect the secondary labels definition

### CODE REVIEW

File: packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx [hunk 1]
```diff
@@ -2,17 +2,20 @@ import * as React from 'react'
...
+import { QueryVisualizationContextType } from '../QueryVisualizationWrapper'
 
 export type DatasetProps = {
   data?: any
   schema?: any
   secondaryLabelLimit?: number
   selectColumns?: SelectColumn[]
   columnModels?: ColumnModel[]
+  genericCardSchema?: GenericCardSchema
+  queryVisualizationContext: QueryVisualizationContextType
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx:18]: The Card already attempts to pass these into the special row renderers, we just need to define them in the props to gain access to them.

File: packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx [hunk 2]
```diff
@@ -31,7 +34,14 @@ class Dataset extends React.Component<DatasetProps, never> {
...
 
   public render() {
-    const { data, schema, selectColumns, columnModels } = this.props
+    const {
+      data,
+      schema,
+      selectColumns,
+      columnModels,
+      genericCardSchema,
+      queryVisualizationContext: { getColumnDisplayName },
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx:43]: Need the queryVisualizationContext to have access to the getColumnDisplayName (which possibly transforms the camel case column name into a more friendly version)

File: packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx [hunk 3]
```diff
@@ -63,6 +73,20 @@ class Dataset extends React.Component<DatasetProps, never> {
       ['SIZE', fileSize],
       ['FILES', fileCount],
     ]
+    if (genericCardSchema && genericCardSchema.secondaryLabels) {
```
Comment (jay-hodgson) [inline packages/synapse-react-client/src/lib/containers/row_renderers/Dataset.tsx:76]: Originally, FUNDER, SIZE, and FILES were the hard-coded secondary values for Datasets.  Now we add the ability to define your own in the configuration (no special rendering of the value today)

**Review by hallieswan — APPROVED** [review]
Looks good!

**Pull request merged**